wavedump/src/analyze-waveforms

```diff
@@ -436,31 +454,52 @@ if __name__ == '__main__':
+                    
+                    temp_charge = integrate(x,y,a,b)
+                    
+                    if args.crosstalk and group == 'sodium' and channel != ct_ch and ch_data[ct_ch]['sodium_peak'] is not None:
+                        ct_ch_chg = np.array(ct_data[ct_ch][len(charge):len(charge)+len(temp_charge)])
+                        ct_511_chg = ch_data[ct_ch]['sodium_peak'][0]
+                        ct_511_std = ch_data[ct_ch]['sodium_peak'][2]
+                        mask_lower = ct_511_chg - ct_thres*ct_511_std < ct_ch_chg
+                        mask_upper = ct_511_chg + ct_thres*ct_511_std > ct_ch_chg
+                        mask = np.logical_and(mask_lower, mask_upper)
+                        if channel not in ct_waveforms:
+                            ct_waveforms[channel] = y[mask,:]
+                        else:
+                            ct_waveforms[channel] = np.append(ct_waveforms[channel], y[mask, :], axis=0)
 
-                    charge.extend(integrate(x,y, a, b))
+                    charge.extend(temp_charge)
+                
+                if args.crosstalk and group == 'sodium':
+                    ct_data[channel] = charge
                         
                 if args.plot or args.print_pdfs:
                     if group == 'sodium':
                         avg_y = ch_data[channel]['avg_pulse_y']
                     else:
-                        # avg_y for the spe waveform is only used for
-                        # plotting
+                        # avg_y for everything else is only used for plotting
                         avg_y = np.mean(y, axis=0)
-                    plot_time_volt(x, y, channel, group, a, b, avg_y=avg_y, pdf=args.print_pdfs)
-                    # Plotting the filtered voltage signal. Doesn't have to
-                    # be included in the final draft of this code.
+                    
+                    plot_time_volt(x, y, channel, group, a, b, avg_y=avg_y, pdf=args.print_pdfs, charges=np.array(temp_charge))
+                    
+                    if args.crosstalk and channel != ct_ch and group == 'sodium':
+                        plot_time_volt(x, ct_waveforms[channel], channel, f'{group} Crosstalk', a, b, avg_y=np.median(ct_waveforms[channel], axis=0), pdf=args.print_pdfs)
+                    # Plotting the filtered voltage signal. Doesn't have to be included in the final draft of this code.
                     if group == 'spe':
                         plot_time_volt(x, high_filter_y, channel, f"high filter {group}", a, b, pdf=args.print_pdfs)
                 
                 ##################
                 # Creating Histogram
                 ##################
-                bins = get_bins(charge, cutoff=200)
+                if group == 'sodium':
+                    bins = get_bins(charge, cutoff=200)
+                else:
+                    bins = get_bins(charge)
```

Comment:
I'm not sure if this is the best place to do it or if we should do it at the database level, but we should probably add something here to make sure that it doesn't create 10000+ bins since that will take up too much space in the database in the end. We might just want to hardcode the bins eventually. We should talk about this.